javalib/src/main/scala/java/util/HashSet.scala

```diff
@@ -0,0 +1,82 @@
+package java.util
+
+import scala.collection.mutable
+import scala.collection.JavaConversions._
+
+class HashSet[E] extends AbstractSet[E] with Set[E]
+                                        with Cloneable
+                                        with Serializable {
+  def this(initialCapacity: Int, loadFactor: Float) =
+    this()
+
+  def this(initialCapacity: Int) =
+    this()
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  protected val inner: mutable.Set[Box[E]] =
+    new mutable.HashSet[Box[E]]()
+
+  override def contains(o: Any): Boolean =
+    inner.contains(Box(o))
```

Comment:
This doesn't compile. You probably need to do `o.asInstanceOf[E]`, just like you did for `ConcurrentHashMap`.